### DISCUSSION TITLE
a few improvement ideas

### DISCUSSION THREAD

**Discussion by raphael-dario** [TOrnelas/SegmentedProgressBar#35] [Ideas]
1. we can't change properties through code `segmentSelectedBackgroundColor` and `segmentBackgroundColor` and even `timePerSegmentMs` have only private setters.
2. `segmentCornerRadius` applies to all segments, although we might want to set corner radius only to start of 1st segment and to end of last segment. currently, I do a trick of containing it in a CardView to achieve this.
3. if we don't use animations (just `pause()` and `setPosition()`) we cannot have the progress fully filled.
```
segmentCount = 10
setPosition(10)
pause()
```
result => the progressbar is empty.

**Comment by EminYen** [comment]
timePerSegmentMs must be public, to set time for segment programmatically and dynamically.
